app/models/product.rb

```diff
@@ -0,0 +1,8 @@
+class Product < ApplicationRecord
+
+  has_many :line_items, dependent: :restrict_with_error
```

Comment:
what to do if I want to stop selling a product?

Reply:
We can add another column to products table storing whether this product is available or not.

Reply:
I have created another table and model BranchProduct storing which store is selling which products.